File: RuntimeData/xnaruntime/RuntimeXNA/RuntimeXNA/Expressions/EXP_GETRGBAT.cs
//----------------------------------------------------------------------------------
//
// COULEUR POINT IMAGE
//
//----------------------------------------------------------------------------------
using System;
using RuntimeXNA.RunLoop;
using RuntimeXNA.Banks;
using RuntimeXNA.Objects;
using RuntimeXNA.Services;
namespace RuntimeXNA.Expressions
{
	
	public class EXP_GETRGBAT:CExpOi
	{
		public override void  evaluate(CRun rhPtr)
		{
			CObject hoPtr = rhPtr.rhEvtProg.get_ExpressionObjects(oiList);
			rhPtr.rh4CurToken++;
			if (hoPtr == null)
			{
                rhPtr.getCurrentResult().forceInt(0);
				return ;
			}
			int x = rhPtr.get_ExpressionInt();
			rhPtr.rh4CurToken++;
			int y = rhPtr.get_ExpressionInt();
			
			int rgb = 0;

            if (hoPtr.roc.rcImage != - 1)
			{
                CImage image = rhPtr.rhApp.imageBank.getImageFromHandle(hoPtr.roc.rcImage);
                if (x > 0 && x < image.width && y > 0 && y < image.height)
                {
                    int[] pixels = new int[image.width * image.height];
                    image.image.GetData(pixels);
                    rgb = pixels[y * image.width + x] & 0xFFFFFF;
                    rgb = CServices.swapRGB(rgb);
                }
			}
            rhPtr.getCurrentResult().forceInt(rgb);
		}
	}
}

File: RuntimeData/runtime/app/src/main/java/Expressions/EXP_GETRGBAT.java
/* Copyright (c) 1996-2013 Clickteam
 *
 * This source code is part of the Android exporter for Clickteam Multimedia Fusion 2.
 * 
 * Permission is hereby granted to any person obtaining a legal copy 
 * of Clickteam Multimedia Fusion 2 to use or modify this source code for 
 * debugging, optimizing, or customizing applications created with 
 * Clickteam Multimedia Fusion 2.  Any other use of this source code is prohibited.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
 * IN THE SOFTWARE.
 */
//----------------------------------------------------------------------------------
//
// COULEUR POINT IMAGE
//
//----------------------------------------------------------------------------------
package Expressions;

import RunLoop.*;
import Objects.*;

public class EXP_GETRGBAT extends CExpOi
{
    public void evaluate(CRun rhPtr)
    {        
	CObject hoPtr=rhPtr.rhEvtProg.get_ExpressionObjects(oiList);
	rhPtr.rh4CurToken++;
	if (hoPtr==null)
	{
	    rhPtr.rh4Results[rhPtr.rh4PosPile].forceInt(0);
	    return;
	}
	int x=rhPtr.get_ExpressionInt();
	rhPtr.rh4CurToken++;
	int y=rhPtr.get_ExpressionInt();

        rhPtr.rh4Results[rhPtr.rh4PosPile].forceInt(rhPtr.getRGBAt(hoPtr, x, y));
    }    
}
